network/network.go

```diff
@@ -132,8 +132,8 @@ type network struct {
 
 	b Builder
 
-	stateLock       sync.Mutex
-	pendingBytes    int
+	stateLock       sync.RWMutex
```

Comment:
Can we document that this stateLock should never be held when grabbing a peer lock?